plugins/modules/nautobot_server.py

```diff
@@ -0,0 +1,391 @@
+                default=False, required=False, type="bool", aliases=["fail_fast"]
+            ),
+            fixtures=dict(default=None, required=False, type="str"),
+            testrunner=dict(
+                default=None, required=False, type="str", aliases=["test_runner"]
+            ),
+            skip=dict(default=None, required=False, type="bool"),
+            merge=dict(default=None, required=False, type="bool"),
+            link=dict(default=None, required=False, type="bool"),
+        ),
+    )
+
+    command = module.params["command"]
+    project_path = module.params["project_path"]
+    virtualenv = module.params["virtualenv"]
+
+    for param in specific_params:
+        value = module.params[param]
+        if param in specific_boolean_params:
+            value = module.boolean(value)
+        if value and param not in command_allowed_param_map[command]:
+            module.fail_json(
+                msg="%s param is incompatible with command=%s" % (param, command)
+            )
+
+    for param in command_required_param_map.get(command, ()):
+        if not module.params[param]:
+            module.fail_json(
+                msg="%s param is required for command=%s" % (param, command)
+            )
+
+    _ensure_virtualenv(module)
+
+    cmd = "nautobot-server %s" % (command,)
+
+    environ_vars = {}
+    if project_path:
+        environ_vars["NAUTOBOT_ROOT"] = project_path
+
+    db_password = module.params["db_password"]
+    if db_password:
+        environ_vars["NAUTOBOT_DB_PASSWORD"] = db_password
+
+    if command in noinput_commands:
+        cmd = "%s --noinput" % cmd
+
+    for param in general_params:
+        if module.params[param]:
+            cmd = "%s --%s=%s" % (cmd, param, module.params[param])
+
+    for param in specific_boolean_params:
+        if module.boolean(module.params[param]):
+            cmd = "%s --%s" % (cmd, param)
+
+    # these params always get tacked on the end of the command
+    for param in end_of_command_params:
+        if module.params[param]:
+            cmd = "%s %s" % (cmd, module.params[param])
+
+    rc, out, err = module.run_command(
+        cmd, cwd=project_path, environ_update=environ_vars
+    )
+    if rc != 0:
+        # Handling expected errors
+        if command == "createcachetable" and "table" in err and "already exists" in err:
+            out = "already exists."
+        elif "createsuperuser" in command and "username is already taken" in err:
+            out = "Admin user already exists."
+        else:
+            # Customize some output messages
+            if "Unknown command:" in err:
+                _fail(module, cmd, err, "Unknown django command: %s" % command)
+            elif "fe_sendauth: no password supplied" in err:
+                _fail(
+                    module,
+                    cmd,
+                    err,
+                    "No DB password provided, you must supply 'db_password' for this command",
```

Comment:
If we still need this and it cannot be used from the `nautobot_config.py`, we should prob try and fail sooner if they're running a specific command that doesn't have a required argument.

Reply:
This should not happen now that we enforce the db_password, but if at some time we remove this requirement, the error helps to identify what's going on.